app/service/data_svc.py

```diff
@@ -49,8 +49,14 @@ def __init__(self, dao):
             await self.dao.create('core_parser', parser)
         return 'Saved ability: %s' % id
 
-    async def create_adversary(self, name, description, phases):
-        identifier = await self.dao.create('core_adversary', dict(name=name.lower(), description=description))
+    async def create_adversary(self, name, description, phases, locked):
+        ident = await self.dao.get('core_adversary', dict(name=name.lower()))
+        if len(ident) == 0:
+            identifier = await self.dao.create('core_adversary', dict(name=name.lower(), description=description, locked=locked))
+        else:
+            if ident[0]['locked']:
+                return await self.create_adversary(name + "[*]", description, phases, locked)
+            identifier = ident[0]['id']
```

Comment:
let's try this without a recursive call. can we

Reply:
This is exactly how this has been handled in the past in CALDERA, but I can get rid of this and punt the fine-tuning of permissions until we deal with user profiles.